### PR TITLE
Support aggregated queries

### PR DESCRIPTION
**Pull request by Avokadoen**
this PR implements queries to child nodes as described in issue #4
To achieve this there is a new package 'localhttp' that create an abstraction for among other things
writing a response from either the local node or its children

### CODE REVIEW

**Comment by Avokadoen** [comment]
Draft until error handling in localhandler is done

**Comment by Avokadoen** [comment]
In the event this gets merged there should be a new issue related to handling not getting a response from a query
See [here](https://github.com/nlnwa/gowarcserver/blob/8f1787eec404b1fd89f8dfaec7cb7d49343215d2/pkg/server/localhttp/localhandler.go) for relevant code section

File: pkg/server/warcserver/resourcehandler.go [hunk 1]
```diff
@@ -31,15 +32,22 @@ import (
...
+	db       *index.DB
+	children *localhttp.Children
 }
 
 func (h *resourceHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	var renderFunc RenderFunc = func(w http.ResponseWriter, record *cdx.Cdx, cdxApi *cdxServerApi) error {
+	localhttp.FirstQuery(h, w, r, time.Second*3)
+}
+
+func (h *resourceHandler) ServeLocalHTTP(wg *sync.WaitGroup, r *http.Request) (*localhttp.Writer, error) {
```
Comment (maeb) [inline pkg/server/warcserver/resourcehandler.go:48]: Unused parameter `wg *sync.WaitGroup`?

Comment (Avokadoen) [reply pkg/server/warcserver/resourcehandler.go:48]: It's used to conform to the [LocalHandler](https://github.com/nlnwa/gowarcserver/pull/24/files#diff-6002aa2ff0dafdb74b46e888c884af7de414ade2c48f2df7710f9b79bbe6ea44) interface

Comment (Avokadoen) [reply pkg/server/warcserver/resourcehandler.go:48]: Woops, your right!

Comment (Avokadoen) [reply pkg/server/warcserver/resourcehandler.go]: Resolved in commit ab8c719

**Comment by Avokadoen** [comment]
TODO: pull master and resolve conflicts

**Comment by maeb** [comment]
Proposed resolve in branch [distribute-3](/nlnwa/gowarcserver/tree/distribute-3).

**Pull request merged**